identities/models.py

```diff
@@ -44,3 +48,74 @@ class Identity(models.Model):
 
     def __str__(self):
         return str(self.id)
+
+    def remove_details(self, user):
+        updated_details = {}
+        for attribute, value in self.details.iteritems():
+            updated_details[attribute] = "removed"
+        self.details = updated_details
+        self.updated_at = datetime.datetime.now()
+        self.updated_by = user
+        self.save()
+
+
+class OptOut(models.Model):
+    """An optout"""
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False,
+                          help_text="UUID this optout request.")
+    identity = models.ForeignKey(Identity, null=True,
+                                 help_text="UUID for the identity opting out.")
```

Comment:
I'm a bit -1 on allowing this to be null and having the post save hook fill it in. I'd rather have a helper method (maybe `.create_for_address(addr_type, addr)` on the manager) look up the id first and then create the object.